src/python/zquantum/core/interfaces/backend.py

```diff
@@ -64,7 +65,7 @@ def run_circuit_and_measure(
 
     def run_circuitset_and_measure(
```

Comment:
What about changing the name of this function as well, so we get rid of `circuitset` for good?

We could sth like this: 
```
def runcircuitset_and_measure(...):
    DepreciationWarning("runcircuitset_and_measure will now be callend XXX")
    return XXX(...)
```
It will also help with in case someone used `circuit_set` as a keyword argument instead of positional.
Not sure what the good name for XXX would be, as `run_circuits_and_measure` is too similar to `run_circuit_and_measure` and could cause confusion.
Perhaps `run_multiple_circuits_and_measure`.

Reply:
According to the docstrings the main use case for having methods with collections is batching. WDYT about specifying this explicitly in the name? Sth like:
```
def run_and_measure(
    self, circuit: AnyCircuit, n_samples: Optional[int] = None, **kwargs
):
    pass

def run_and_measure_batch(
    self, circuits_batch: Iterable[AnyCircuit], n_samples: Optional[int] = None, **kwargs
):
    pass

def get_expectation_values(
    self, circuit: AnyCircuit, operator: SymbolicOperator, **kwargs
) -> ExpectationValues:
    pass

def get_expectation_values_batch(
    self, circuits_batch: Iterable[AnyCircuit], operator: SymbolicOperator, **kwargs
) -> List[ExpectationValues]:
    pass

Reply:
Yeah, I like using the `batch` word.
CC: @max-radin @simonwa7 – sounds good?

Reply:
Not sure if this is the time/place for this, but `run_circuit_and_measure` should also be deprecated as it's just an instance of running a batch of 1

Reply:
I would also be in favor of just: 
```
def run_and_measure(
    self, circuits: Iterable[AnyCircuit], n_samples: Optional[int] = None, **kwargs
):
    pass

def get_expectation_values(
    self, circuit:  Iterable[AnyCircuit], operator: SymbolicOperator, **kwargs
) -> ExpectationValues:
    pass
```

Reply:
I think @simonwa7 is right – both in how it should be handled and that it might not be time to do all that right now and we should get back to it later.

Reply:
There are concerns against doing too much changes at once (see my comment below the PR). @dexter2206 and I would suggest to do the renaming on some other occasion.